### PR TITLE
refactor(Scripts): Update vanilla scripts to std::chrono

### PR DESCRIPTION
**Pull request by Maelthyrr**
## Changes Proposed:
-  Use std::chrono in all vanilla scripts

## Issues Addressed:

- Part of https://github.com/azerothcore/azerothcore-wotlk/issues/6490

## Tests Performed:

- 
- 

## Known Issues and TODO List:

- [ ] Some methods doesn't accept std::chrono for the moment (repeartevent, getnextevent, despawn...)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_feralas.cpp [hunk 1]
```diff
@@ -207,8 +207,8 @@ class npc_shay_leafrunner : public CreatureScript
                 me->SetFaction(FACTION_ESCORT_N_NEUTRAL_ACTIVE);
                 me->GetMotionMaster()->MoveFollow(player, 3.f, M_PI);
 
-                _events.ScheduleEvent(EVENT_WANDERING_START, urand(40 * IN_MILLISECONDS, 70 * IN_MILLISECONDS));
-                _events.ScheduleEvent(EVENT_CHECK_FOLLOWER, 30 * IN_MILLISECONDS);
+                _events.ScheduleEvent(EVENT_WANDERING_START, 40s, 70s);
```
Comment (Helias) [inline src/server/scripts/Kalimdor/zone_feralas.cpp:210]: `urand(x,y)` is the same to `x,y` ? you are adding a third parameter to `ScheduleEvent` is it the same?

Comment (Nyeriah) [reply src/server/scripts/Kalimdor/zone_feralas.cpp:210]: > `urand(x,y)` is the same to `x,y` ? you are adding a third parameter to `ScheduleEvent` is it the same?

Yes.

File: src/server/scripts/EasternKingdoms/AlteracValley/alterac_valley.cpp [hunk 2]
```diff
@@ -95,12 +95,12 @@ class npc_av_marshal_or_warmaster : public CreatureScript
...
         {
             events.Reset();
-            events.ScheduleEvent(EVENT_CHARGE_TARGET, urand(2 * IN_MILLISECONDS, 12 * IN_MILLISECONDS));
-            events.ScheduleEvent(EVENT_CLEAVE, urand(1 * IN_MILLISECONDS, 11 * IN_MILLISECONDS));
-            events.ScheduleEvent(EVENT_DEMORALIZING_SHOUT, 2000);
-            events.ScheduleEvent(EVENT_WHIRLWIND, urand(5 * IN_MILLISECONDS, 20 * IN_MILLISECONDS));
-            events.ScheduleEvent(EVENT_ENRAGE, urand(5 * IN_MILLISECONDS, 20 * IN_MILLISECONDS));
-            events.ScheduleEvent(EVENT_CHECK_RESET, 5000);
+            events.ScheduleEvent(EVENT_CHARGE_TARGET, 2s, 12s);
+            events.ScheduleEvent(EVENT_CLEAVE, 1s,11s);
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/AlteracValley/alterac_valley.cpp:99]:
```suggestion
            events.ScheduleEvent(EVENT_CLEAVE, 1s, 11s);
```

**Review by Nyeriah — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_arlokk.cpp [hunk 3]
```diff
@@ -258,14 +258,14 @@ class boss_arlokk : public CreatureScript
                             AttackStart(target);
                         me->RemoveAura(SPELL_SUPER_INVIS);
                         me->RemoveAura(SPELL_VANISH);
-                        events.ScheduleEvent(EVENT_RAVAGE, urand(10000, 14000), 0, PHASE_TWO);
-                        events.ScheduleEvent(EVENT_TRANSFORM_BACK, urand(30000, 40000), 0, PHASE_TWO);
+                        events.ScheduleEvent(EVENT_RAVAGE, 10s, 14s, 0, PHASE_TWO);
+                        events.ScheduleEvent(EVENT_TRANSFORM_BACK, 30s, 40s), 0, PHASE_TWO);
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_arlokk.cpp:262]:
```suggestion
                        events.ScheduleEvent(EVENT_TRANSFORM_BACK, 30s, 40s, 0, PHASE_TWO);
```

**Comment by Nyeriah** [comment]
> Some methods doesn't accept std::chrono for the moment (repeartevent, getnextevent, despawn...)

@Maelthyrr use Repeat()

**Pull request merged**